File: ClientApp/app/components/northwind/order.ts
import { Shipper } from './shipper';
import { Employee } from './employee';

export class Order
{
    orderID: number;
    CustomerID: string;
    EmployeeID: number;
    orderDate: string;
    requiredDate: string;
    shippedDate: string;
    shipVia: string;
    freight: number;
    shipName: string;
    shipAddress: string;
    shipCity: string;
    shipRegion: string;
    shipPostalCode: string;
    shipCountry: string;

    shipper: Shipper;
    employee: Employee;

    toString(): string {
        return JSON.stringify(this);
    }
}

File: Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace NorthwindAngular.Models
{
    public class Order
    {
        public Order()
        {
            this.OrderDetails = new HashSet<OrderDetail>();
        }

        public int OrderID { get; set; }
        public string CustomerID { get; set; }

        public Nullable<int> EmployeeID { get; set; }

        public Nullable<DateTime> OrderDate { get; set; }
        public Nullable<DateTime> RequiredDate { get; set; }
        public Nullable<DateTime> ShippedDate { get; set; }

        [ForeignKey("Shipper")]
        public Nullable<int> ShipVia { get; set; }

        public Nullable<decimal> Freight { get; set; }
        public string ShipName { get; set; }
        public string ShipAddress { get; set; }
        public string ShipCity { get; set; }
        public string ShipRegion { get; set; }
        public string ShipPostalCode { get; set; }
        public string ShipCountry { get; set; }

        public virtual ICollection<OrderDetail> OrderDetails { get; set; }
        //public virtual Customer Customer { get; set; }

        public virtual Employee Employee { get; set; }
        public virtual Shipper Shipper { get; set; }
    }
}
